cpp/src/arrow/type.h

```diff
@@ -2120,35 +2108,4 @@ ARROW_EXPORT
 std::string ToString(TimeUnit::type unit);
 
 }  // namespace internal
-
-// Helpers to get instances of data types based on general categories
-
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& SignedIntTypes();
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& UnsignedIntTypes();
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& IntTypes();
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& FloatingPointTypes();
-// Number types without boolean
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& NumericTypes();
-// Binary and string-like types (except fixed-size binary)
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& BaseBinaryTypes();
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& BinaryTypes();
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& StringTypes();
-// Temporal types including time and timestamps for each unit
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& TemporalTypes();
-// Interval types
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& IntervalTypes();
-// Integer, floating point, base binary, and temporal
-ARROW_EXPORT
-const std::vector<std::shared_ptr<DataType>>& PrimitiveTypes();
-
```

Comment:
There wasn't really a good reason for these functions to be here -- they probably even should be in an internal namespace in arrow::compute